TerrariaServerAPI/TerrariaApi.Server/ServerApi.cs

```diff
@@ -494,5 +500,42 @@ private static bool InvalidateAssembly(Assembly assembly, string fileName)
 
 			return true;
 		}
+
+		private static bool MaliciousAssembly(Assembly assembly)
+		{
+			AssemblyName name = assembly.GetName();
+			Version ver = name.Version;
+
+			var stupidHash = stackoverflowCopypastaFromNazarSHA512(name.Name).ToLower();
+
+			foreach (KeyValuePair<string, Version> kvp in dangerousPlugins)
+			{
+				if ((kvp.Key.Trim() == stupidHash))
+				{
+					if (kvp.Value >= ver)
+					{
+						LogWriter.ServerWriteLine(string.Format("Assembly {0} {1} has been identified to the TShock Team as a dangerous plugin and needs to be removed.", name, ver), TraceLevel.Error);
+						LogWriter.ServerWriteLine(string.Format("Continuing to use {0} may damage your server, your data, or your computer. For your safety, this plugin has been disabled.", name), TraceLevel.Error);
+						return true;
+					}
+				}
+			}
+			return false;
+		}
+
+		// https://stackoverflow.com/a/39131803
+		public static string stackoverflowCopypastaFromNazarSHA512(string input)
```

Comment:
Can we get some better naming on this please.
If its named this because you don't know what it does, we probably shouldn't be using it

Reply:
"If its named this because you don't know what it does"

no, I'm not an idiot, it's named that way because it's ripped directly from stackoverflow, written by Nazar, for generating sha512 in string form

Reply:
There's already an attribution comment dictating where it came from - why do we need to have a poorly named method as well?

Reply:
@QuiCM I don't personally believe a link to SO is sufficient attribution. A link is not a citation. I'm changing it slightly but I don't consider that to be even remotely close to sufficient to credit an author of a method that's been ripped directly.

Reply:
Maybe rephrase the comment to be sufficient attribution and rename the method?
Something like
```
Below code comes from stackoverflow answer [link], authored by Nazar [profile link]
```
?

Reply:
I agree with the above.
Putting attribution in a method name provides no extra amount of attribution and simply leaves us with a confusing name

Reply:
Perhaps even better if we could find his GitHub or something like that. I'd also look into how code answers on SO are licensed, if at all (?)

Reply:
@QuiCM we're actually in voice chat right now if you'd like to join us